### PR TITLE
PACA-426: Allow empty strings in enum values

### PR DESCRIPTION
**Pull request by brianburton**
**JIRA Ticket:**
[PACA-426](https://jira.cms.gov/browse/PACA-426)

**User Story or Bug Summary:**

Many RDA enum fields have the option of sending an unknown value rather than a known enum. The enum fields are fixed, and we know the length of those, however the unknown values could be of a different length (however it can't be larger since the RDA's DB schema doesn't support that). We should remove the min length field check, so we can allow values, like the empty string, to be persisted.

---

### What Does This PR Do?

Changes enum string value validation to allow an arbitrary minimum length instead of hard coding it to one.  Changes transformers to use `minLength = 0` where appropriate.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.
* Are there any enum fields that were missed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist
I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/McsClaimTransformer.java [hunk 1]
```diff
@@ -1042,7 +1042,7 @@ private PreAdjMcsLocation transformMessageImpl(
     transformer.copyEnumAsString(
         namePrefix + PreAdjMcsLocation.Fields.idrLocActvCode,
         true,
-        1,
+        0,
```
Comment (KyRobbins) [inline apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/McsClaimTransformer.java:1045]: I'm not totally sure about what this number is, but based on how many were updated, should it be a constant value?

Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/McsClaimTransformer.java:1045]: Previously the `minLength` defaulted to 1.  Now most enums need to use `0` instead.  But that's a good point.  I'll make the default `0` and just pass in `1` in the few places that need it.

Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/McsClaimTransformer.java:1045]: That's the minimum allowed string length.

Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/McsClaimTransformer.java]: I have refactored the code to eliminate the explicit zeros.

**Review by KyRobbins — APPROVED** [review]

**Review by ericbuckley — APPROVED** [review]

**Review by christrotteradhoc — APPROVED** [review]

**Pull request merged**